crates/neon-runtime/src/neon_task.h

```diff
@@ -59,7 +59,9 @@ class Task {
     }
 
     v8::Local<v8::Function> callback = v8::Local<v8::Function>::New(isolate_, callback_);
-    callback->Call(context, v8::Null(isolate_), 2, argv);
+    node::MakeCallback(isolate_, context->Global(), callback, 2, argv);
+    callback_.Reset();
+    context_.Reset();
```

Comment:
These missing `Reset` on the `Persistent` objects were a memory leak in the existing code.

Reply:
The changes look okay to me and MakeCallback() is the right method to call but apropos those resource leaks, their destructors should clean out the references.

Are you saying that isn't happening or is there something that is holding onto the Task longer than necessary?

Reply:
I'm not sure their destructors clean-out the references. They are wrapped in in `Nan` with a destructor added to call this.

You can observe the leak by taking a heap dump, calling an async API a bunch, forcing a GC, and then taking another heapdump.

Reply:
https://github.com/nodejs/nan/blob/b2455ac52e97568456089c5658bbd64a016112b8/nan.h#L1383-L1385

Reply:
Oh, I think I see the issue now - those `v8::Persistent` members should have been of type `v8::Global` or `v8::Persistent<T,M>` with `M::kResetInDestructor = true`.  Might be good to check if the code base has more of these issues.